addons/promote/common/src/main/java/org/commonjava/indy/promote/metrics/PathGauges.java

```diff
@@ -0,0 +1,81 @@
+package org.commonjava.indy.promote.metrics;
+
+import com.codahale.metrics.Gauge;
+import org.commonjava.indy.metrics.IndyMetricsManager;
+import org.commonjava.indy.promote.model.PathsPromoteResult;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@ApplicationScoped
+public class PathGauges
+{
+    @Inject
+    private IndyMetricsManager metricsManager;
+
+    private AtomicInteger total = new AtomicInteger();
```

Comment:
Would it make sense to use actual Gauge instances here, so we get the histogram features that they implement? This might help us keep more accurate rate information that isn't connected to the rate at which promotions happen (the update method), or the rate at which the reporter runs (the lambda implementations in the registerPathPromotionGauges method now).

Reply:
Previously we run registerPathPromotionGauges at the rate at which promotion runs but that did not work. This is because that the metricRegistry.gauge() only add new Gauges when it is not there. If exists, it just returns the one in use. So what we got is just the first some Gauges created by the first promotion attempt. I.e, we cannot create many Gauges with the same name. Gauges like singleton object and it report current values when asked by the reporter.

Reply:
yeah, I was thinking we'd still only register a single set of instances, but we'd set them to fields in this class and then update them directly...